### PR TITLE
[ADD] art_craft: The Arts & Crafts Shop

### PR DESCRIPTION
**Pull request by jva-odoo**
This module provide a colourful array of paints, brushes, paper, beads, tools,
colourful trays and other crafting (Handmade) essentials. They are a treasure
trove for anyone seeking to express their creativity, explore new hobbies,
or create beautiful handmade items.

Co-authored-by: Jigar Vaghela <jva@odoo.com>
Co-authored-by: Gurpreet Singh <gusi@odoo.com>

### CODE REVIEW

**Comment by paso-odoo** [comment]
@jva-odoo 

Tested in 17.0

File: art_craft/demo/purchase_order_line.xml [hunk 1]
```diff
@@ -0,0 +1,159 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="purchase_order_line_1" model="purchase.order.line">
+        <field name="order_id" ref="art_craft.purchase_order_1"/>
+        <field name="name">[WPL NANDI/s] 3D Corrugated Nandi (S)</field>
+        <field name="product_qty">10.0</field>
+        <field name="product_id" ref="art_craft.product_product_8"/>
+        <field name="price_unit">50.8</field>
```
Comment (jva-odoo) [inline art_craft/demo/purchase_order_line.xml:8]:
```suggestion
        <field name="name">[WPL NANDI/s] 3D Corrugated Nandi (S)</field>
        <field name="product_id" ref="art_craft.product_product_8"/>
        <field name="product_qty">10.0</field>
        <field name="price_unit">50.8</field>
        <field name="order_id" ref="art_craft.purchase_order_1"/>
```
change sequence

File: art_craft/demo/purchase_order_post.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            obj().search([('state', '=', 'assigned'), ('move_ids', 'in',
+                (
+                    obj().env.ref('art_craft.purchase_order_1') +
+                    obj().env.ref('art_craft.purchase_order_10') +
+                    obj().env.ref('art_craft.purchase_order_11') +
+                    obj().env.ref('art_craft.purchase_order_12') +
+                    obj().env.ref('art_craft.purchase_order_14') +
+                    obj().env.ref('art_craft.purchase_order_15')
+                ).group_id.stock_move_ids.ids)
+            ]).ids"/>
```
Comment (jva-odoo) [inline art_craft/demo/purchase_order_post.xml:34]:
```suggestion
            (
                    obj().env.ref('art_craft.purchase_order_1') +
                    obj().env.ref('art_craft.purchase_order_10') +
                    obj().env.ref('art_craft.purchase_order_11') +
                    obj().env.ref('art_craft.purchase_order_12') +
                    obj().env.ref('art_craft.purchase_order_14') +
                    obj().env.ref('art_craft.purchase_order_15')
                ).group_id.stock_move_ids.picking_id.ids"/>
```

File: art_craft/demo/res_partner.xml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        <field name="name">Anital Oliver</field>
+        <field name="email">anita.oliver00@example.com</field>
+        <field name="mobile">(538)-672-3185</field>
+        <field name="city">Pleasant Hill</field>
+        <field name="state_id" ref="base.state_us_5"/>
+        <field name="country_id" ref="base.us"/>
+        <field name="title" ref="base.res_partner_title_miss"/>
+        <field name="property_stock_supplier" ref="stock.stock_location_suppliers"/>
+        <field name="property_stock_customer" ref="stock.stock_location_customers"/>
+        <field name="customer_rank">4</field>
```
Comment (jva-odoo) [inline art_craft/demo/res_partner.xml:12]:
```suggestion
```

File: art_craft/demo/res_partner.xml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    <record id="res_partner_10" model="res.partner">
+        <field name="name">Anital Oliver</field>
+        <field name="email">anita.oliver00@example.com</field>
+        <field name="mobile">(538)-672-3185</field>
+        <field name="city">Pleasant Hill</field>
+        <field name="state_id" ref="base.state_us_5"/>
+        <field name="country_id" ref="base.us"/>
+        <field name="title" ref="base.res_partner_title_miss"/>
+        <field name="property_stock_supplier" ref="stock.stock_location_suppliers"/>
+        <field name="property_stock_customer" ref="stock.stock_location_customers"/>
```
Comment (jva-odoo) [inline art_craft/demo/res_partner.xml:11]:
```suggestion
```
not needed

File: art_craft/demo/sale_order.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        <field name="website_id" ref="website.default_website"/>
+    </record>
+    <record id="sale_order_4" model="sale.order">
+        <field name="partner_id" ref="art_craft.res_partner_19"/>
+        <field name="delivery_rating_success" eval="True"/>
+        <field name="website_id" ref="website.default_website"/>
+    </record>
+    <record id="sale_order_5" model="sale.order">
+        <field name="partner_id" ref="art_craft.res_partner_10"/>
+        <field name="effective_date">2023-08-16 09:13:59</field>
```
Comment (jva-odoo) [inline art_craft/demo/sale_order.xml:24]:
```suggestion
```
compute base on delivery

File: art_craft/demo/sale_order_post.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    </function>
+
+    <function name="button_validate" model="stock.picking" context="{'skip_sms': True}">
+        <value model="stock.picking" eval="
+            obj().search([('state', '=', 'assigned'), ('move_ids', 'in',
+                (
+                    obj().env.ref('art_craft.sale_order_5')
+                ).procurement_group_id.stock_move_ids.ids)
+            ]).ids"/>
+    </function>
```
Comment (jva-odoo) [inline art_craft/demo/sale_order_post.xml:17]: same as above

File: art_craft/demo/purchase_order_post.xml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+            obj().search([('state', '=', 'assigned'), ('id', 'in',
+                (
+                    obj().env.ref('art_craft.purchase_order_1') +
+                    obj().env.ref('art_craft.purchase_order_10') +
+                    obj().env.ref('art_craft.purchase_order_11') +
+                    obj().env.ref('art_craft.purchase_order_12') +
+                    obj().env.ref('art_craft.purchase_order_14') +
+                    obj().env.ref('art_craft.purchase_order_15')
+                ).group_id.stock_move_ids.picking_id.ids)
+            ]).ids"/>
```
Comment (jva-odoo) [inline art_craft/demo/purchase_order_post.xml:34]:
```suggestion
        <value model="stock.picking" eval="
             (
                    obj().env.ref('art_craft.purchase_order_1') +
                    obj().env.ref('art_craft.purchase_order_10') +
                    obj().env.ref('art_craft.purchase_order_11') +
                    obj().env.ref('art_craft.purchase_order_12') +
                    obj().env.ref('art_craft.purchase_order_14') +
                    obj().env.ref('art_craft.purchase_order_15')
            ).group_id.stock_move_ids.picking_id.ids"/>
```

File: art_craft/data/product_pricelist.xml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="product_pricelist_2" model="product.pricelist">
+        <field name="name">B2B Pricelist</field>
+        <field name="website_id" ref="website.default_website"/>
+        <field name="discount_policy">without_discount</field>
+    </record>
+    <record id="product_pricelist_3" model="product.pricelist">
+        <field name="name">Default pricelist B2C</field>
+        <field name="website_id" eval="False"/>
```
Comment (jva-odoo) [inline art_craft/data/product_pricelist.xml:10]:
```suggestion
```

File: art_craft/data/product_pricelist_item.xml [hunk 9]
```diff
@@ -0,0 +1,54 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
```
Comment (jva-odoo) [inline art_craft/data/product_pricelist_item.xml:2]:
```suggestion
<odoo noupdate="1">
```

File: art_craft/demo/website_theme_apply.xml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+        <value  model="ir.module.module" eval="obj().env.ref('website.default_website')"/>
+    </function>
+    <function model="ir.ui.view" name="write">
+        <value model="ir.ui.view" eval="obj().env['website'].with_context(website_id=obj().env.ref('website.default_website').id).viewref('website.homepage').id"/>
+        <value model="ir.ui.view" eval="{'arch': obj().env.ref('art_craft.homepage').arch}"/>
+    </function>
+    <function model="ir.ui.view" name="write">
+        <value model="ir.ui.view" eval="obj().env['website'].with_context(website_id=obj().env.ref('website.default_website').id).viewref('website.about-us').id"/>
+        <value model="ir.ui.view" eval="{'arch': obj().env.ref('art_craft.about-us').arch}"/>
+    </function>
```
Comment (jva-odoo) [inline art_craft/demo/website_theme_apply.xml:18]: `website.about-us` not need to write it's have page

File: art_craft/demo/website_views.xml [hunk 11]
```diff
@@ -0,0 +1,758 @@
...
+                            </div>
+                        </section>
+                    </div>
+                </t>
+            </t>
+        </field>
+    </record>
+    <record id="art_craft.contactus" model="ir.ui.view">
+        <field name="name">Contact Us</field>
+        <field name="key">website.contactus</field>
```
Comment (jva-odoo) [inline art_craft/demo/website_views.xml:318]:
```suggestion
        <field name="key">art_craft.contactus</field>
```

File: art_craft/demo/website_views.xml [hunk 12]
```diff
@@ -0,0 +1,758 @@
...
+                            </div>
+                        </section>
+                    </div>
+                </t>
+            </t>
+        </field>
+    </record>
+    <record id="art_craft.about-us" model="ir.ui.view">
+        <field name="name">About Us</field>
+        <field name="key">website.about-us</field>
```
Comment (jva-odoo) [inline art_craft/demo/website_views.xml:580]:
```suggestion
        <field name="key">art_craft.about-us</field>
```

File: art_craft/data/product_product.xml [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+    <record id="product_product_34" model="product.product">
+        <field name="product_tmpl_id" ref="art_craft.product_template_26"/>
+    </record>
+    <record id="product_product_6" model="product.product">
+        <field name="product_tmpl_id" ref="art_craft.product_template_6"/>
+    </record>
+    <record id="product_product_8" model="product.product">
+        <field name="product_tmpl_id" ref="art_craft.product_template_8"/>
+        <field name="default_code">WPL NANDI/s</field>
+        <field name="standard_price">50.800000000000004</field>
```
Comment (jva-odoo) [inline art_craft/data/product_product.xml:111]:
```suggestion
        <field name="standard_price">50.80</field>
```

**Pull request merged**